tests/test_admin.py

```diff
@@ -1481,6 +1483,55 @@ def test_unpublish_view_can_be_accessed_by_get_request(self):
         # no status change has been tracked
         self.assertEqual(StateTracking.objects.all().count(), 0)
 
+    def test_unpublish_view_uses_setting_to_populate_context(self):
+        poll_version = factories.PollVersionFactory(state=constants.PUBLISHED)
+        url = self.get_admin_url(
+            self.versionable.version_model_proxy, "unpublish", poll_version.pk
+        )
+
+        def unpublish_context1(request, version, *args, **kwargs):
+            return "Don't unpublish cats. Seriously."
+
+        def unpublish_context2(request, version, *args, **kwargs):
+            return "Unpublish the mice instead."
+
+        def publish_context(request, version, *args, **kwargs):
+            return "Publish cat pictures only. People aren't interested in anything else."
+
+        versioning_ext = apps.get_app_config('djangocms_versioning').cms_extension
+        extra_context = OrderedDict({
+            'unpublish': {'cats': unpublish_context1, 'mice': unpublish_context2},
```

Comment:
The other way round, it's the dicts inside which need to be OrderedDict objects.